### PR TITLE
app-misc/solaar: Version bump to 1.0.1, Python 3.7 and EAPI 7

### PR DESCRIPTION
**Pull request by cjmayo**
Bug: https://bugs.gentoo.org/691208
Package-Manager: Portage-2.3.69, Repoman-2.3.16
Signed-off-by: Chris Mayo <aklhfex@gmail.com>

---

Update web addresses and swap gnome2-utils for xdg.

```diff
--- solaar-0.9.2-r3.ebuild
+++ solaar-1.0.1.ebuild
@@ -1,18 +1,18 @@
 # Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
-PYTHON_COMPAT=( python{2_7,3_5,3_6} )
+EAPI=7
+PYTHON_COMPAT=( python2_7 python3_{5,6,7} )
 
-inherit distutils-r1 udev user linux-info gnome2-utils
+inherit distutils-r1 linux-info udev user xdg
 
 DESCRIPTION="A Linux device manager for Logitech's Unifying Receiver peripherals"
-HOMEPAGE="https://pwr.github.com/Solaar/"
-SRC_URI="https://github.com/pwr/Solaar/archive/${PV}.tar.gz -> ${P}.tar.gz"
+HOMEPAGE="https://pwr-solaar.github.io/Solaar/"
+SRC_URI="https://github.com/pwr-Solaar/Solaar/archive/${PV}.tar.gz -> ${P}.tar.gz"
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="amd64 ~arm x86"
+KEYWORDS="~amd64 ~arm ~x86"
 IUSE="doc"
 
 RDEPEND=">=dev-python/pyudev-0.13[${PYTHON_USEDEP}]
@@ -51,8 +51,5 @@
 		elog "Users must be in the plugdev group to use this application."
 	fi
 
-	gnome2_icon_cache_update
+	xdg_pkg_postinst
 }
-
-pkg_preinst() { gnome2_icon_savelist; }
-pkg_postrm() { gnome2_icon_cache_update; }
```

### CODE REVIEW

**Comment by Katana-Steel** [comment]
tried it on my local repo, LGTM too.

Picked up the lightspeed connctor right away..

File: app-misc/solaar/solaar-1.0.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	udev_dorules rules.d/*.rules
+
+	dodoc docs/devices.md
+	if use doc; then
+		dodoc -r docs/*
+	fi
+}
+
+pkg_postinst() {
+	enewgroup plugdev
```
Comment (juippis) [inline app-misc/solaar/solaar-1.0.1.ebuild]: My only concern is this. It should be done via the new GLEP-81 way. Also it allows you to removed the replacing_versions part below.

I'm a bit amazed and confused no one has pushed acct-group/plugdev yet...

Comment (lovesegfault) [reply app-misc/solaar/solaar-1.0.1.ebuild]: Nice catch @juippis, completely forgot about GLEP-81

**Comment by cjmayo** [comment]
GLEP 81 does say:
"The old method may still be used for existing users/groups, in existing packages."
https://www.gentoo.org/glep/glep-0081.html#id5

I haven't got time to propose acct-group/plugdev myself, but if it gets added I'm happy to update this PR.

**Comment by juippis** [comment]
> "The old method may still be used for existing users/groups, in existing packages."
> https://www.gentoo.org/glep/glep-0081.html#id5

**existing** packages. Meaning, old packages won't need to be updated (it would require a revbump for each one).
But _new_ packages should follow GLEP-81.

**Comment by cjmayo** [comment]
To me this is a revbump of an existing package, not a new package, using an existing group that has already been created by many ebuilds.
Not to say that using the new system wouldn't be a good thing.

**Comment by cjmayo** [comment]
acct-group/plugdev now available, updated to use it and remove REPLACING_VERSIONS (indeed whole pkg_postinst) as suggested.

**Pull request closed without merging**